sources/editor/Stride.Editor/Engine/EntityExtensions.cs

```diff
@@ -20,12 +20,41 @@ public static Entity FindSubEntity(this Entity entity, Guid subEntityId)
             if (entity.Id == subEntityId)
                 return entity;
 
-            return entity.GetChildren().DepthFirst(x => x.GetChildren()).FirstOrDefault(x => x.Id == subEntityId);
+            foreach (var child in entity.Transform.Children)
+            {
+                if (child.Entity.FindSubEntity(subEntityId) is Entity e)
```

Comment:
Incorrect. You replaced a depth-first search which is sequential with a recursive call which could potentially crash the stack. Deep entity hierarchies will crash the editor.

Reply:
Threads have by default around one meg of stack space, let's say that half of it is already taken, somehow, and this recursion allocs around 50 bytes per call, you could recurse a hierarchy with a depth of 10000 items before crashing, of course all of that is napkin math but I'm being safe here, it's very unlikely that half of the stack is already taken and even more unlikely that each recursion takes that much of the stack. AFAIK Quantum has a couple of place were it has to recurse, that's likely the first place where it will crash because of deep hierarchies. If I end up rewriting quantum I'll solve them both

Reply:
Not true at all. We had a test scene (from a real project) that could very easily reach that limit.

Reply:
A real project that contains entities whose furthest ancestor is 'easily' 10000 generations deep ? Sorry but I don't believe that's likely, perhaps at runtime but within the game studio it's just not manageable and it doesn't make a lot of sense to have such a construct in a game either, what kind of ridiculous scenario could possibly require 10000 parents.
Anyway tested that inside a build of the gamestudio before those changes and the process silently crashes before it can even bring the main window up, gamestudio just can't load anything deeper than around 1250 on my end.

If you still don't think those recursions are safe I could conditionally swap back to the previous version through ``RuntimeHelpers.ProbeForSufficientStack``